be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala

```diff
@@ -25,10 +25,10 @@ final case class Channel(channel: String) {
    *
    * @return an array of bytes corresponding to the decoded laoId or None if an error occurred
    */
-  def decodeChannelLaoId: Option[Array[Byte]] = channel match {
+  def decodeChannelLaoId: Option[Base64Data] = channel match {
     case _ if channel.startsWith(Channel.ROOT_CHANNEL_PREFIX) =>
       Try(Base64.getUrlDecoder.decode(channel.substring(Channel.ROOT_CHANNEL_PREFIX.length).split(Channel.SEPARATOR)(0).getBytes)) match {
-        case Success(value) => Some(value)
+        case Success(value) => Some(Base64Data.encode(value))
```

Comment:
I believe the LaoID you're getting here as "value" is already the base64-representation of the channel.
Why do you need to encode it (i.e. get the base64 representation of that string), as opposed to just doing the following ? 
```suggestion
        case Success(value) => Some(Base64Data(value))
```

Reply:
It is a mistake, I fixed it, thank you!

Reply:
However, I had to modify the Try code above for that